girder/api/describe.py

```diff
@@ -598,14 +598,50 @@ def wrapped(*args, **kwargs):
                 sortdir = kwargs.pop('sortdir', None) or kwargs['params'].pop('sortdir', None)
                 kwargs['sort'] = [(kwargs['sort'], sortdir)]
 
-            return fun(*args, **kwargs)
+            result = fun(*args, **kwargs)
+            self._addPagingHeaders(kwargs, result)
+            return result
 
         if self.hide:
             wrapped.description = None
         else:
             wrapped.description = self.description
         return wrapped
 
+    def _addPagingHeaders(self, params, result):
+        """
+        Inject a "Link" header when this is a paged request.
+        Follows: https://tools.ietf.org/html/rfc5988
+
+        :param params: An object containing the paging parameters
+        :param result: The list of results from query
+        """
+        if not self.description.hasPagingParams or \
+           not cherrypy.request.method == 'GET':
+            return
+
+        qs = cherrypy.lib.httputil.parse_query_string(cherrypy.request.query_string)
+        offset = params['offset']
+        limit = params['limit']
+        link = []
+
+        qs['offset'] = 0
+        link.append('<%s>; rel="first"' % cherrypy.url(qs=qs))
```

Comment:
Now that I think of it, I'm not sure this will provide the correct link when proxying the API server unless the proxy rewrites the link header.  Should I use the configured API root here instead?